file_watcher/lastrun_file_monitor.py

```diff
@@ -0,0 +1,271 @@
+        self.latest_cycle = self.get_latest_cycle()
+
+        # Database setup and checks if runs missed then recovery
+        self.db_updater = DBUpdater(ip=db_ip, username=db_username, password=db_password)
+        self.latest_known_run_from_db = self.get_latest_run_from_db()
+        logger.info("Last run in DB is: %s", self.latest_known_run_from_db)
+        if self.latest_known_run_from_db is None:
+            logger.info("Adding latest run to DB as there is no data: %s", self.last_recorded_run_from_file)
+            self.update_db_with_latest_run(self.last_recorded_run_from_file)
+            self.latest_known_run_from_db = self.last_recorded_run_from_file
+
+    async def _init(self) -> None:
+        """
+        This function needs to be called before any other function and is the equivalent of a setup, it has to be
+        done outside __init__ because it is an Async function, and async functionality cannot be completed inside
+        __init__.
+        """
+        if (
+            self.latest_known_run_from_db is not None
+            and self.last_recorded_run_from_file is not None
+            and int(self.latest_known_run_from_db) < int(self.last_recorded_run_from_file)
+        ):
+            logger.info(
+                "Recovering lost runs between %s and%s", self.last_recorded_run_from_file, self.latest_known_run_from_db
+            )
+            if self.latest_known_run_from_db is not None and self.last_recorded_run_from_file is not None:
+                await self.recover_lost_runs(self.latest_known_run_from_db, self.last_recorded_run_from_file)
+            self.latest_known_run_from_db = self.last_recorded_run_from_file
+
+    def get_latest_run_from_db(self) -> Union[str, None]:
+        """
+        Retrieve the latest run from the database
+        :return: Return the latest run for the instrument that is set on this object
+        """
+        # This likely contains NDX<INSTNAME> so remove the NDX and go for it with the DB
+        actual_instrument = self.instrument[3:]
+        return self.db_updater.get_latest_run(actual_instrument)
+
+    async def watch_for_new_runs(self, run_once: bool = False) -> None:
+        """
+        This is the main loop for waiting for new runs and triggering
+        :param run_once: Defaults to False, and will only run the loop once, aimed at simplicity for testing.
+        """
+        logger.info("Starting watcher...")
+        run = True
+        while run:
+            if run_once:
+                run = False
+            current_time = datetime.datetime.now()
+            time_between_cycle_folder_checks = current_time - self.last_cycle_folder_check
+            # If it's been 6 hours do another check for the latest folder
+            if time_between_cycle_folder_checks.total_seconds() > 21600:
+                self.latest_cycle = self.get_latest_cycle()
+                self.last_cycle_folder_check = current_time
```

Comment:
```suggestion
            self._check_for_new_cycle_folder()
```
I think it would aid future readability to extract this into a private method

Reply:
Good idea